pjsip/src/pjsip-ua/sip_inv.c

```diff
@@ -698,13 +698,20 @@ static pj_bool_t mod_inv_on_rx_request(pjsip_rx_data *rdata)
             }
 
             /* Now we can terminate the INVITE transaction */
-            pj_assert(inv->invite_tsx->status_code >= 200);
-            pjsip_tsx_terminate(inv->invite_tsx, 
-                                inv->invite_tsx->status_code);
+
+            /* In the case of an INVITE transaction, if the response was not
+             * 2xx, the ACK is considered part of the transaction, so
+             * should be handled by the transaction.
+             */
+            if (inv->invite_tsx->status_code/100 == 2) {
+                pjsip_tsx_terminate(inv->invite_tsx,
+                                    inv->invite_tsx->status_code);
+            }
```

Comment:
If the tsx status code is 3xx and the remote wrongly put a bad Via branch in its ACK (i.e: not equal to the branch in its INVITE) so the ACK has to be handled here (by the invite session), but by this patch, the INVITE tsx will be no longer terminated immediately (will be terminated later by tsx timeout timer?).

Not sure if it is a problem (note that there should be only one INVITE tsx at any time), but perhaps it is better if it can be avoided (maintaining the current behavior). So an idea, what if the `pjsip_tsx_terminate()` is moved to a timer instead (and use the tsx group lock for the timer scheduling to avoid tsx premature destroy)?